### PR TITLE
Rename Target to Asset and ScanResult to AssetScan

### PR DESCRIPTION
**Pull request by sambetts-cisco**
## Description

This aligns with the UX and also aligns better with the objects' purposes. This is a breaking change with no migration, VMClarity will need to be reinstalled after this change.

## Type of Change

[ ] Bug Fix  
[ ] New Feature  
[X] Breaking Change  
[X] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [X] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [X] Existing issues have been referenced (where applicable)
- [X] I have verified this change is not present in other open pull requests
- [X] Functionality is documented
- [X] All code style checks pass
- [X] New code contribution is covered by automated tests
- [X] All new and existing tests pass

### CODE REVIEW

File: api/openapi.yaml [hunk 1]
```diff
@@ -894,8 +894,8 @@ components:
           $ref: '#/components/schemas/ScanConfigRelationship'
         scanConfigSnapshot:
           $ref: '#/components/schemas/ScanConfigData'
-        targetIDs:
-          description: List of target IDs that are targeted for scanning as part of this scan
+        assetIDs:
+          description: List of asset IDs that are asseted for scanning as part of this scan
```
Comment (akpsgit) [inline api/openapi.yaml:898]: That is a good one XD

File: api/openapi.yaml [hunk 2]
```diff
@@ -1351,37 +1351,37 @@ components:
...
+          description: Total number of scans that have ever run for this asset
           type: integer
         summary:
           $ref: '#/components/schemas/ScanFindingsSummary'
 
-    Target:
+    Asset:
       type: object
-      description: Describes a target object.
+      description: Describes a asset object.
```
Comment (akpsgit) [inline api/openapi.yaml:1382]: an asset

File: api/openapi.yaml [hunk 3]
```diff
@@ -1395,36 +1395,36 @@ components:
...
               #
               # readOnly: true
-          # required: ['targetInfo']
+          # required: ['assetInfo']
 
-    TargetRelationship:
+    AssetRelationship:
       type: object
-      description: Describes a relationship to a target which can be expanded.
+      description: Describes a relationship to a asset which can be expanded.
```
Comment (akpsgit) [inline api/openapi.yaml:1402]: same

File: backend/pkg/database/gorm/target.go [hunk 4]
```diff
@@ -30,90 +30,90 @@ import (
...
+		return models.Asset{}, &common.BadRequestError{
+			Reason: "can not specify id field when creating a new Asset",
 		}
 	}
 
-	// Check that targetInfo was provided by the user, it's a required field for a target.
-	if target.TargetInfo == nil {
-		return models.Target{}, &common.BadRequestError{
-			Reason: "targetInfo is a required field",
+	// Check that assetInfo was provided by the user, it's a required field for a asset.
```
Comment (akpsgit) [inline backend/pkg/database/gorm/target.go:108]: same

File: backend/pkg/database/gorm/target.go [hunk 5]
```diff
@@ -127,161 +127,161 @@ func (t *TargetsTableHandler) CreateTarget(target models.Target) (models.Target,
...
+		return models.Asset{}, &common.BadRequestError{
+			Reason: "id is required to save asset",
 		}
 	}
 
-	// Check that targetInfo was provided by the user, it's a required field for a target.
-	if target.TargetInfo == nil {
-		return models.Target{}, &common.BadRequestError{
-			Reason: "targetInfo is a required field",
+	// Check that assetInfo was provided by the user, it's a required field for a asset.
```
Comment (akpsgit) [inline backend/pkg/database/gorm/target.go:170]: same

File: backend/pkg/database/gorm/target.go [hunk 6]
```diff
@@ -127,161 +127,161 @@ func (t *TargetsTableHandler) CreateTarget(target models.Target) (models.Target,
...
 	}
 
 	switch info := discriminator.(type) {
 	case models.VMInfo:
-		var targets []Target
-		// In the case of creating or updating a target, needs to be checked whether other target exists with same InstanceID and Location.
-		filter := fmt.Sprintf("id ne '%s' and targetInfo/instanceID eq '%s' and targetInfo/location eq '%s'", *target.Id, info.InstanceID, info.Location)
-		err = ODataQuery(t.DB, targetSchemaName, &filter, nil, nil, nil, nil, nil, true, &targets)
+		var assets []Asset
+		// In the case of creating or updating a asset, needs to be checked whether other asset exists with same InstanceID and Location.
```
Comment (akpsgit) [inline backend/pkg/database/gorm/target.go:268]: same

File: runtime_scan/pkg/scanner/scanner.go [hunk 7]
```diff
@@ -177,11 +177,11 @@ func (s *Scanner) Scan(ctx context.Context) {
...
 }
 
-func (s *Scanner) SetTargetScanStatusCompletionError(ctx context.Context, scanResultID, errMsg string) error {
+func (s *Scanner) SetAssetScanStatusCompletionError(ctx context.Context, assetScanID, errMsg string) error {
 	// Get the status and set the completion error
-	status, err := s.backendClient.GetScanResultStatus(ctx, scanResultID)
+	status, err := s.backendClient.GetAssetScanStatus(ctx, assetScanID)
 	if err != nil {
-		return fmt.Errorf("failed to get a target scan status: %v", err)
+		return fmt.Errorf("failed to get a asset scan status: %v", err)
```
Comment (akpsgit) [inline runtime_scan/pkg/scanner/scanner.go:184]: same

File: runtime_scan/pkg/types/types.go [hunk 8]
```diff
@@ -19,11 +19,11 @@ import (
 	"context"
 )
 
-// Job represents a scan process of a target.
+// Job represents a scan process of a asset.
```
Comment (akpsgit) [inline runtime_scan/pkg/types/types.go:22]: same

File: backend/pkg/database/gorm/asset_scan.go [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+func (s *AssetScansTableHandler) GetAssetScans(params models.GetAssetScansParams) (models.AssetScans, error) {
+	var scanResults []AssetScan
+	err := ODataQuery(s.DB, assetAssetScansSchemaName, params.Filter, params.Select, params.Expand, params.OrderBy, params.Top, params.Skip, true, &scanResults)
+	if err != nil {
+		return models.AssetScans{}, err
+	}
+
+	items := make([]models.AssetScan, len(scanResults))
+	for i, scanResult := range scanResults {
+		var tsr models.AssetScan
```
Comment (akpsgit) [inline backend/pkg/database/gorm/asset_scan.go:59]: in all this file need to tsr -> as

File: backend/pkg/database/gorm/asset_scan.go [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+}
+
+func (db *Handler) AssetScansTable() types.AssetScansTable {
+	return &AssetScansTableHandler{
+		DB: db.DB,
+	}
+}
+
+func (s *AssetScansTableHandler) GetAssetScans(params models.GetAssetScansParams) (models.AssetScans, error) {
+	var scanResults []AssetScan
```
Comment (akpsgit) [inline backend/pkg/database/gorm/asset_scan.go:51]: in all this file need to scanResults -> assetScans and  scanResult -> assetScan

File: runtime_scan/pkg/orchestrator/orchestrator.go [hunk 11]
```diff
@@ -35,20 +35,20 @@ type Orchestrator interface {
...
-	config              *_config.OrchestratorConfig
-	scanConfigWatcher   *configwatcher.ScanConfigWatcher
-	scopeDiscoverer     *discovery.ScopeDiscoverer
-	scanResultProcessor *scanresultprocessor.ScanResultProcessor
-	scanWatcher         *scanwatcher.Watcher
-	cancelFunc          context.CancelFunc
+	config             *_config.OrchestratorConfig
+	scanConfigWatcher  *configwatcher.ScanConfigWatcher
+	scopeDiscoverer    *discovery.ScopeDiscoverer
+	assetScanProcessor *scanresultprocessor.AssetScanProcessor
```
Comment (akpsgit) [inline runtime_scan/pkg/orchestrator/orchestrator.go:41]: need to rename the package to assetscanprocessor

File: runtime_scan/pkg/orchestrator/scanresultprocessor/common.go [hunk 12]
```diff
@@ -60,14 +60,14 @@ func (srp *ScanResultProcessor) newerExistingFindingTime(ctx context.Context, ta
 	return found, newerTime, nil
 }
 
-func (srp *ScanResultProcessor) invalidateOlderFindingsByType(ctx context.Context, findingType string, targetID string, completedTime time.Time) error {
+func (srp *AssetScanProcessor) invalidateOlderFindingsByType(ctx context.Context, findingType string, assetID string, completedTime time.Time) error {
 	// Invalidate any findings of this type for this asset where foundOn is
 	// older than this scan result, and has not already been invalidated by
```
Comment (akpsgit) [inline runtime_scan/pkg/orchestrator/scanresultprocessor/common.go:65]: asset scan

File: runtime_scan/pkg/orchestrator/scanresultprocessor/common.go [hunk 13]
```diff
@@ -60,14 +60,14 @@ func (srp *ScanResultProcessor) newerExistingFindingTime(ctx context.Context, ta
...
 	// a scan result older than this scan result.
```
Comment (akpsgit) [inline runtime_scan/pkg/orchestrator/scanresultprocessor/common.go:66]: same

File: backend/pkg/rest/target_controller.go [hunk 14]
```diff
@@ -28,137 +28,137 @@ import (
 	"github.com/openclarity/vmclarity/runtime_scan/pkg/utils"
 )
```
Comment (akpsgit) [inline backend/pkg/rest/target_controller.go:30]: need to rename the file

File: runtime_scan/pkg/cloudinit/cloudinit_template.go [hunk 15]
```diff
@@ -46,11 +46,11 @@ write_files:
           --server {{ .ServerAddress }} \
           --wait-for-server-attached \
           --mount-attached-volume \
-          --scan-result-id {{ .ScanResultID }} \
+          --scan-result-id {{ .AssetScanID }} \
```
Comment (akpsgit) [inline runtime_scan/pkg/cloudinit/cloudinit_template.go:49]: --asset-scan-id

File: cli/cmd/root.go [hunk 16]
```diff
@@ -151,7 +151,7 @@ func init() {
 	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.vmclarity.yaml)")
 	rootCmd.PersistentFlags().StringVar(&output, "output", "", "set output directory path. Stdout is used if not set.")
 	rootCmd.PersistentFlags().StringVar(&server, "server", "", "VMClarity server to export scan results to, for example: http://localhost:9999/api")
-	rootCmd.PersistentFlags().StringVar(&scanResultID, "scan-result-id", "", "the ScanResult ID to export the scan results to")
+	rootCmd.PersistentFlags().StringVar(&assetScanID, "scan-result-id", "", "the AssetScan ID to export the scan results to")
```
Comment (akpsgit) [inline cli/cmd/root.go:154]: same

File: cli/cmd/root.go [hunk 17]
```diff
@@ -151,7 +151,7 @@ func init() {
...
 	rootCmd.PersistentFlags().BoolVar(&mountVolume, "mount-attached-volume", false, "discover for an attached volume and mount it before the scan")
 	rootCmd.PersistentFlags().BoolVar(&waitForServerAttached, "wait-for-server-attached", false, "wait for the VMClarity server to attach the volume")
```
Comment (akpsgit) [inline cli/cmd/root.go:157]: also below

File: cli/pkg/presenter/vmclarity.go [hunk 18]
```diff
@@ -36,28 +36,28 @@ import (
...
 
-	scanResultID models.ScanResultID
+	assetScanID models.AssetScanID
 }
 
 func (v *VMClarityPresenter) ExportSbomResult(ctx context.Context, res *results.Results, famerr families.RunErrors) error {
-	scanResult, err := v.client.GetScanResult(ctx, v.scanResultID, models.GetScanResultsScanResultIDParams{})
+	assetScan, err := v.client.GetAssetScan(ctx, v.assetScanID, models.GetAssetScansAssetScanIDParams{})
 	if err != nil {
 		return fmt.Errorf("failed to get scan result: %w", err)
```
Comment (akpsgit) [inline cli/pkg/presenter/vmclarity.go:50]: same

File: shared/pkg/backendclient/client.go [hunk 19]
```diff
@@ -399,105 +399,105 @@ func (b *BackendClient) GetScans(ctx context.Context, params models.GetScansPara
 }
 
 //nolint:cyclop
-func (b *BackendClient) PostTarget(ctx context.Context, target models.Target) (*models.Target, error) {
-	resp, err := b.apiClient.PostTargetsWithResponse(ctx, target)
+func (b *BackendClient) PostAsset(ctx context.Context, asset models.Asset) (*models.Asset, error) {
+	resp, err := b.apiClient.PostAssetsWithResponse(ctx, asset)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create a target: %v", err)
+		return nil, fmt.Errorf("failed to create a asset: %v", err)
```
Comment (akpsgit) [inline shared/pkg/backendclient/client.go:405]: also in this file "a asset"

File: ui/src/layout/Findings/LinkesDisplay.js [hunk 20]
```diff
@@ -6,7 +6,7 @@ const LinksDisplay = ({title, links}) => (
         <TitleValueDisplay title={title} withOpen defaultOpen>
             {
                 links?.map((link, index) => (
-                    <div key={index}><a href={link} target="_blank" rel="noopener noreferrer">{link}</a></div>
+                    <div key={index}><a href={link} asset="_blank" rel="noopener noreferrer">{link}</a></div>
```
Comment (akpsgit) [inline ui/src/layout/Findings/LinkesDisplay.js:9]: :(

**Comment by akpsgit** [comment]
@sambetts-cisco There are many files that still have "scan result/s". Can you please look for those (i didn't leave a comment in all places.)

File: api/openapi.yaml (hunk 1 above)
Comment (sambetts-cisco) [reply api/openapi.yaml:995]: fixed

File: backend/pkg/database/gorm/asset.go [hunk 21]
```diff
@@ -0,0 +1,316 @@
...
+	// checking the uniqueness.
+	//
+	// We might also be able to do this without locking the table by doing
+	// a single query which includes the uniqueness check like:
+	//
+	// INSERT INTO scan_configs(data) SELECT * FROM (SELECT "<encoded json>") AS tmp WHERE NOT EXISTS (SELECT * FROM scan_configs WHERE JSON_EXTRACT(`Data`, '$.Name') = '<name from input>') LIMIT 1;
+	//
+	// This should return 0 affected fields if there is a conflicting
+	// record in the DB, and should be treated safely by the DB without
+	// locking the table.
```
Comment (ramizpolic) [inline backend/pkg/database/gorm/asset.go:131]: @sambetts-cisco just checking if this is going to impact anything

Comment (sambetts-cisco) [reply backend/pkg/database/gorm/asset.go:131]: This is an existing TODO, not something introduced by this PR (the file was just renamed, target changed to asset). We need to revisit some of these in the future.

File: runtime_scan/pkg/orchestrator/assetscanprocessor/vulnerabilities.go [hunk 22]
```diff
@@ -63,9 +63,9 @@ func (srp *ScanResultProcessor) reconcileResultVulnerabilitiesToFindings(ctx con
 	logger.Infof("Found %d existing vulnerabilities findings for this scan", len(existingMap))
 	logger.Debugf("Existing vulnerabilities map: %v", existingMap)
 
-	if scanResult.Vulnerabilities != nil && scanResult.Vulnerabilities.Vulnerabilities != nil {
+	if assetScan.Vulnerabilities != nil && assetScan.Vulnerabilities.Vulnerabilities != nil {
 		// Create new findings for all the found vulnerabilities
-		for _, vuln := range *scanResult.Vulnerabilities.Vulnerabilities {
+		for _, vuln := range *assetScan.Vulnerabilities.Vulnerabilities {
```
Comment (ramizpolic) [inline runtime_scan/pkg/orchestrator/assetscanprocessor/vulnerabilities.go:68]: Just a note for future here, `assetScan.Vulnerabilities.Vulnerabilities` seems cumbersome, would be good to replace it with something more straightforward.

**Review by ramizpolic — APPROVED** [review]
lgtm, good stuff! 🚀

File: runtime_scan/pkg/orchestrator/config.go [hunk 23]
```diff
@@ -55,12 +55,12 @@ const (
 	ScanReconcileTimeout = "SCAN_RECONCILE_TIMEOUT"
 	ScanTimeout          = "SCAN_TIMEOUT"
 
-	ScanResultPollingInterval  = "SCAN_RESULT_POLLING_INTERVAL"
-	ScanResultReconcileTimeout = "SCAN_RESULT_RECONCILE_TIMEOUT"
-	ScanResultAbortTimeout     = "SCAN_RESULT_ABORT_TIMEOUT"
+	AssetScanPollingInterval  = "SCAN_RESULT_POLLING_INTERVAL"
+	AssetScanReconcileTimeout = "SCAN_RESULT_RECONCILE_TIMEOUT"
+	AssetScanAbortTimeout     = "SCAN_RESULT_ABORT_TIMEOUT"
```
[the remark this answers is not in the document]
Comment (sambetts-cisco) [reply runtime_scan/pkg/orchestrator/config.go:60]: Great spot! these slipped through in the merge conflict this morning. I've fixed in the latest revision.

**Review by ramizpolic — APPROVED** [review]

**Pull request merged**